src/Npgsql/Replication/ReplicationConnection.cs

```diff
@@ -623,7 +634,7 @@ async Task DropReplicationSlotInternal()
                 await
 #endif
                 using var registration = cancellationToken.CanBeCanceled
-                    ? cancellationToken.Register(c => ((ReplicationConnection)c!).Cancel(), this)
+                    ? cancellationToken.Register(c => ((ReplicationConnection)c!).Connector.CancelRequest(), this)
                     : default;
```

Comment:
You changed this at some other place so let's change it here too.
If I understand it correctly we don't have to check for `cancellationToken.CanBeCanceled` here.
Also let's just pass in the connector we have at hands anyways.
```suggestion
                using var registration = cancellationToken.Register(c => ((NpgsqlConnector)c!).CancelRequest(), connector);
```

Reply:
Makes sense. The CanBeCanceled check really makes sense only in very hot paths (and even there I'm not so sure it matters).

Reply:
> and even there I'm not so sure it matters

It's pretty close to a no-op for default tokens.
See https://github.com/dotnet/runtime/blob/14926991738435e28dbe5b21d0c7e253b2fff060/src/libraries/System.Private.CoreLib/src/System/Threading/CancellationToken.cs#L301